ServerSide/yoty/Core/Managers/Bids/BidsManager.cs

```diff
@@ -404,18 +405,31 @@ private IEnumerable<BidEntity> GetSortBids(IEnumerable<BidEntity> bids, BidsSort
 
         public async Task<Response> VoteForSupplier(VotingRequest votingRequest)
         {
-            BidEntity bid = await _context.Bids.Where(b => b.Id == votingRequest.BidId).Include(b => b.CurrentProposals).FirstOrDefaultAsync().ConfigureAwait(false);
+            BidEntity bid = await _context.Bids.Where(b => b.Id == votingRequest.BidId).Include(b => b.CurrentProposals).Include(b => b.CurrentParticipancies).FirstOrDefaultAsync().ConfigureAwait(false);
             if (bid == null)
             {
                 return new Response() { IsOperationSucceeded = false, SuccessOrFailureMessage = BidNotFoundFailString };
             }
 
-            IEnumerable<SupplierProposalEntity> propsalsToUpdate = bid.CurrentProposals.Where(propsal => votingRequest.VotedSuppliersIds.Contains(propsal.SupplierId));
-            foreach (SupplierProposalEntity proposal in propsalsToUpdate)
+            ParticipancyEntity participancy = bid.CurrentParticipancies.Where(p => p.BuyerId == votingRequest.BuyerId).FirstOrDefault();
+            if (participancy == null)
+            {
+                return new Response() { IsOperationSucceeded = false, SuccessOrFailureMessage = $"Buyer {votingRequest.BuyerId} is not part of the bids buyers." };
+            }
+            else if (participancy.HasVoted)
+            {
+                // TODO enable change after vote
+                return new Response() { IsOperationSucceeded = false, SuccessOrFailureMessage = $"Buyer {votingRequest.BuyerId} has already voted" };
+            }
+
+            SupplierProposalEntity proposal = bid.CurrentProposals.Where(p => p.SupplierId == votingRequest.VotedSupplierId).FirstOrDefault();
+            if (proposal == null)
             {
-                proposal.Votes += 1;
+                return new Response() { IsOperationSucceeded = false, SuccessOrFailureMessage = $"no proposal for supplier: {votingRequest.VotedSupplierId} has found for the bid." };
             }
 
+            participancy.HasVoted = true;
+            proposal.Votes += 1;
```

Comment:
The DB needs to be updated @tzachihakmon 
Add:

try
            {
                _context.Set<ParticipancyEntity>().Update(participancy)
                _context.Set<SupplierProposalEntity>().Update(proposal)
                await _context.SaveChangesAsync().ConfigureAwait(false);
            }
            catch (Exception ex)
            {
                return new Response() { IsOperationSucceeded = false, SuccessOrFailureMessage = ex.Message };
            }